lib/data_explorer_query_group_bookmarkable.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class DataExplorerQueryGroupBookmarkable < BaseBookmarkable
+  def self.model
+    DataExplorer::QueryGroup
+  end
+
+  def self.serializer
+    UserDataExplorerQueryGroupBookmarkSerializer
+  end
+
+  def self.preload_associations
+    [:data_explorer_queries, :groups]
+  end
+
+  def self.list_query(user, guardian)
+    group_ids = user.groups.pluck(:id) 
```

Comment:
I think `user.visible_groups` would be better here, that also covers the case of a user still being in a group but the group's security settings changing later on.

Reply:
Done.